src/core/config/config.go

```diff
@@ -204,84 +188,49 @@ func TokenPrivateKeyPath() string {
 
 // LDAPConf returns the setting of ldap server
 func LDAPConf() (*models.LdapConf, error) {
-	cfg, err := mg.Get()
+	err := cfgMgr.Load()
 	if err != nil {
 		return nil, err
 	}
-	ldapConf := &models.LdapConf{}
-	ldapConf.LdapURL = utils.SafeCastString(cfg[common.LDAPURL])
-	ldapConf.LdapSearchDn = utils.SafeCastString(cfg[common.LDAPSearchDN])
-	ldapConf.LdapSearchPassword = utils.SafeCastString(cfg[common.LDAPSearchPwd])
-	ldapConf.LdapBaseDn = utils.SafeCastString(cfg[common.LDAPBaseDN])
-	ldapConf.LdapUID = utils.SafeCastString(cfg[common.LDAPUID])
-	ldapConf.LdapFilter = utils.SafeCastString(cfg[common.LDAPFilter])
-	ldapConf.LdapScope = int(utils.SafeCastFloat64(cfg[common.LDAPScope]))
-	ldapConf.LdapConnectionTimeout = int(utils.SafeCastFloat64(cfg[common.LDAPTimeout]))
-	if cfg[common.LDAPVerifyCert] != nil {
-		ldapConf.LdapVerifyCert = cfg[common.LDAPVerifyCert].(bool)
-	} else {
-		ldapConf.LdapVerifyCert = true
-	}
-
-	return ldapConf, nil
+	return &models.LdapConf{
+		LdapURL:               cfgMgr.Get(common.LDAPURL).GetString(),
+		LdapSearchDn:          cfgMgr.Get(common.LDAPSearchDN).GetString(),
+		LdapSearchPassword:    cfgMgr.Get(common.LDAPSearchPwd).GetString(),
+		LdapBaseDn:            cfgMgr.Get(common.LDAPBaseDN).GetString(),
+		LdapUID:               cfgMgr.Get(common.LDAPUID).GetString(),
+		LdapFilter:            cfgMgr.Get(common.LDAPFilter).GetString(),
+		LdapScope:             cfgMgr.Get(common.LDAPScope).GetInt(),
+		LdapConnectionTimeout: cfgMgr.Get(common.LDAPTimeout).GetInt(),
+		LdapVerifyCert:        cfgMgr.Get(common.LDAPVerifyCert).GetBool(),
+	}, nil
 }
 
 // LDAPGroupConf returns the setting of ldap group search
 func LDAPGroupConf() (*models.LdapGroupConf, error) {
-
-	cfg, err := mg.Get()
-	if err != nil {
-		return nil, err
-	}
-
-	ldapGroupConf := &models.LdapGroupConf{LdapGroupSearchScope: 2}
-	if _, ok := cfg[common.LDAPGroupBaseDN]; ok {
-		ldapGroupConf.LdapGroupBaseDN = utils.SafeCastString(cfg[common.LDAPGroupBaseDN])
-	}
-	if _, ok := cfg[common.LDAPGroupSearchFilter]; ok {
-		ldapGroupConf.LdapGroupFilter = utils.SafeCastString(cfg[common.LDAPGroupSearchFilter])
-	}
-	if _, ok := cfg[common.LDAPGroupAttributeName]; ok {
-		ldapGroupConf.LdapGroupNameAttribute = utils.SafeCastString(cfg[common.LDAPGroupAttributeName])
-	}
-	if _, ok := cfg[common.LDAPGroupSearchScope]; ok {
-		if scopeStr, ok := cfg[common.LDAPGroupSearchScope].(string); ok {
-			ldapGroupConf.LdapGroupSearchScope, err = strconv.Atoi(scopeStr)
-		}
-		if scopeFloat, ok := cfg[common.LDAPGroupSearchScope].(float64); ok {
-			ldapGroupConf.LdapGroupSearchScope = int(scopeFloat)
-		}
-	}
-	if _, ok := cfg[common.LdapGroupAdminDn]; ok {
-		ldapGroupConf.LdapGroupAdminDN = cfg[common.LdapGroupAdminDn].(string)
-	}
-	return ldapGroupConf, nil
+	return &models.LdapGroupConf{
```

Comment:
Why some methods call the `Load()` and others don't?

Reply:
Before accessing user settings, need to call Load()
For system settings, no need to call Load()